client/pdo/client/controller/commands/create.py

```diff
@@ -134,17 +137,28 @@ def command_create(state, bindings, pargs) :
     logger.info('Loaded contract code for %s', contract_class)
 
     # ---------- set up the enclave clients ----------
-    try :
+    enclaveclients = []
+    enclave_names = state.get(['Service', 'EnclaveServiceNames'], [])
+    if len(enclave_names) > 0: #use the database to get the list of enclaves for the contract
+        logger.info('Using eservice database to look up service URL for the contract enclave')
+        try:
+            db = ServiceDB_Manager(service_type='eservice', file_name=state.get(['Service', 'EnclaveServiceDatabaseFile']))
+            for name in enclave_names:
```

Comment:
i'm curious as to why you push the service db manager at this low level? why isn't it a persistent object that is re-used throughout the interaction?